crates/stages/src/stages/bodies.rs

```diff
@@ -346,29 +366,43 @@ mod tests {
         assert!(runner.validate_execution(input, output.ok()).is_ok(), "execution validation");
     }
 
-    /// Checks that the stage asks to unwind if pre-validation of the block fails.
+    /// Checks that the stage returns to the pipeline on validation failure.
     #[tokio::test]
     async fn pre_validation_failure() {
         let (stage_progress, previous_stage) = (1, 20);
 
         // Set up test runner
         let mut runner = BodyTestRunner::default();
+
         let input = ExecInput {
             previous_stage: Some((PREV_STAGE_ID, previous_stage)),
             stage_progress: Some(stage_progress),
         };
-        runner.seed_execution(input).expect("failed to seed execution");
+        let blocks = runner.seed_execution(input).expect("failed to seed execution");
 
         // Fail validation
-        runner.consensus.set_fail_validation(true);
+        let responses = blocks
+            .iter()
+            .map(|b| {
+                (
+                    b.hash(),
+                    Err(DownloadError::BlockValidation {
+                        hash: b.hash(),
+                        error: consensus::Error::BaseFeeMissing,
+                    }),
+                )
+            })
+            .collect::<HashMap<_, _>>();
+        runner.set_responses(responses);
```

Comment:
why is this change needed?

Reply:
because the consensus checks were moved into the downloader and the `TetsConsensus` isn't actually called in the test downloader. we just set the responses

Reply:
:+1: